app/controllers/api/offers_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class Api::OffersController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    offer = Offer.create(check_and_lump_parameters)
+    if offer.persisted?
+      render json: { message: 'Your offer has been sent!' }
+    else
+      render json: { message: offer.errors.messages.to_a.flatten.join(' ').capitalize }, status: 422
+    end
+  rescue StandardError => e
+    render json: { message: e.message }, status: 422
+  end
+
+  private
+
+  def check_and_lump_parameters
```

Comment:
Dont understand the lump....

Reply:
My thought was that it was lumping all the params together, as some come as params and one as headers. Changed the name now to check_and_convert_parameters instead, might be more descriptive